cookbooks/aws-parallelcluster-environment/recipes/config/shared_storages.rb

```diff
@@ -50,6 +51,7 @@
     options node['cluster']['nfs']['hard_mount_options']
     retries 10
     retry_delay 6
+    only_if { ::File.directory?("/opt/intel") }
```

Comment:
I know this is not your fault, but this condition looks quite silly: we mount `/opt/intel` only if the `/opt/intel` folder already exists. Personally I would be happier if we put `only_if { !arm_instance? }` here.

Also, I just noticed the way the `volume` resource was implemented imposes the exported dir and the mount point to be the exact same path, which will annoy us very soon I believe.

Reply:
This is the previous code guard and I restored this on purpose since it worked.

It may be that `only_if { !arm_instance? }` works, but it will cover only a corner case and it may easily break if tomorrow we add another one.
And unless we have a centralized point where we clearly express the conditions when this folder is created (that we can also use as documentation) I prefer this implementation.

As for the `volume` resource implementation .. I only used it. I'm still not sure it is really useful since it is nearly a simple wrapper of an existing resource ..but let's give it a try...